src/main/java/com/ecommerce/Controller/CartController.java

```diff
@@ -0,0 +1,57 @@
+package com.ecommerce.Controller;
+
+import com.ecommerce.Entity.Products;
+import com.ecommerce.ExceptionHandler.ProductNotFoundException;
+import com.ecommerce.ExceptionHandler.UserNotFoundException;
+import com.ecommerce.Service.CartService;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1")
+public class CartController {
+
+    @Autowired
+    CartService cartService;
+
+    @GetMapping("/cart/user/{userId}")
+    public ResponseEntity<List<Products>> getCartContents(@PathVariable Long userId) {
+        try {
+            List<Products> cartContents = cartService.getCartContents(userId);
+            return ResponseEntity.ok(cartContents);
+        } catch (UserNotFoundException e) {
```

Comment:
Do you have a global exception handler? If yes then throw the exception form your service, if not you can refactor this later